app/models/course.rb

```diff
@@ -17,6 +17,16 @@ class Course < ActiveRecord::Base
               message: 'should not contain white spaces'
             }
 
+  validates :title,
+            presence: true,
+            length: { within: 1..40 }
+
+  validates :material_url,
+            format: {
+                without: /:/,
+                message: 'should not contain colon (:). Remove "http://" from the beginning if you have it'
```

Comment:
yeah, this doesn't sound like a reasonable request?
And since we can always preprocess the input; its more user friendly if the user can just copy/paste the url from his/her browser.